### PR TITLE
Created Abstract Model

### PR DESCRIPTION
**Pull request by skramar**
## JIRA

* [Main JIRA ticket]()

## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/abstractmodel/models.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+from django.db import models
+from utils.utils import LOGGER
+
+class Abstract(models.Model):
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: maybe it is better to call this class like `AbstractModel`? It is more meaningly =)

File: eventually/abstractmodel/models.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from django.db import models
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: Add module docrsting

File: eventually/abstractmodel/models.py (hunk 1 above)
Comment (redmyt) [inline eventually/abstractmodel/models.py]: Add class docstring

File: eventually/abstractmodel/models.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    def __repr__(self):
+        """
+        This magic method is redefined to show class and id of object.
+
+        :return: class, id
+
+        """
+        return f'{self.__class__.__name__}(id={self.id})'
+
+    def to_dict():
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: maybe we have to use `@abc.abstractmethod` decorator?

File: eventually/abstractmodel/models.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        :return: object instance or None if object does not exist
+
+        """
+        try:
+            obj = cls.objects.get(id=obj_id)
+            return obj
+        except cls.DoesNotExist:
+                LOGGER.error(f'Certain {cls.__name__} does not exist')
+
+    def create():
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: maybe we have to use `@abc.abstractmethod` decorator?

File: eventually/abstractmodel/models.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        """
+        Create object.
+        Must be overloaded.
+
+        """
+        return ValueError
+
+    def update():
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: maybe we have to use `@abc.abstractmethod` decorator?

File: eventually/abstractmodel/models.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        :type obj_id: integer
+
+        :return: object instance or None if object does not exist
+
+        """
+        try:
+            obj = cls.objects.get(id=obj_id)
+            return obj
+        except cls.DoesNotExist:
+                LOGGER.error(f'Certain {cls.__name__} does not exist')
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: add more sense to the log message. We can use id method parameter.

File: eventually/abstractmodel/models.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        :return: True if object seccessfully deleted or
+                 None if object does not exist
+
+        """
+        try:
+            obj = cls.objects.get(id=obj_id)
+            obj.delete()
+            return True
+        except (cls.DoesNotExist, AttributeError):
+            LOGGER.error(f'Certain {cls.__name__} does not exist')
```
Comment (redmyt) [inline eventually/abstractmodel/models.py]: add more sense to the log message. We can use id method parameter.

File: eventually/abstractmodel/views.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
```
Comment (redmyt) [inline eventually/abstractmodel/views.py]: do we need this file?

File: eventually/comment/models.py [hunk 9]
```diff
@@ -5,16 +5,16 @@
 This module implements class that represents the comment entity.
```
Comment (redmyt) [inline eventually/comment/models.py:5]: maybe it is a good idea to refactor existing models in the other branches?

**Review by redmyt — CHANGES_REQUESTED** [review]

File: eventually/abstractmodel/models.py (hunk 2 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 1 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 3 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 6 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 4 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 5 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/models.py (hunk 7 above)
Comment (skramar) [reply eventually/abstractmodel/models.py]: done

File: eventually/abstractmodel/views.py (hunk 8 above)
Comment (skramar) [reply eventually/abstractmodel/views.py]: done

File: eventually/comment/models.py (hunk 9 above)
Comment (skramar) [reply eventually/comment/models.py:5]: done

**Review by redmyt — APPROVED** [review]

**Pull request merged**